File: problems/b/babelfish/babelfish.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <bits/stdc++.h>

using namespace std;

int main() {
    map<string, string> dictionary;

    string line;
    do {
        getline(cin, line);

        stringstream ss(line);
        string english, foreign_language;
        ss >> english >> foreign_language;

        dictionary[foreign_language] = english;
    } while (line.size() != 0);

    string word;
    while (cin >> word) {
        if (dictionary.find(word) == dictionary.end()) {
            cout << "eh" << endl;
        }
        else {
            cout << dictionary[word] << endl;
        }
    }
}

File: problems/b/babelfish/babelfish.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""

def main():
    dictionary = dict()
    while True:
        line = input()
        if line:
            english, foreign_language = line.split()
            dictionary[foreign_language] = english
        else:
            break

    while True:
        try:
            word = input()
            if word:
                print(dictionary.get(word, "eh"))
        except EOFError:
            break
    

if __name__ == '__main__':
    main()